stan/math/rev/core/arena_allocator.hpp

```diff
@@ -13,33 +13,68 @@ namespace math {
 template <typename T>
 struct arena_allocator {
   using value_type = T;
+  using pointer = T*;
+  using const_pointer = const T*;
+  using reference = T&;
+  using const_reference = const T&;
+  using size_type = size_t;
+  using difference_type = ptrdiff_t;
+  using propagate_on_container_move_assignment = std::true_type;
+
+  constexpr arena_allocator() noexcept {};
+
+  template <class U>
+  constexpr arena_allocator(const arena_allocator<U>&) noexcept {}
+
+  template <class U>
+  constexpr arena_allocator<T>& operator=(const arena_allocator<U>&) noexcept {
+    return *this;
+  }
```

Comment:
We don't need to explicitely define these, as the implementations are the same as defaults.